plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -162,6 +172,16 @@ public IcebergSplitSource(
                 tableHandle.getUnenforcedPredicate());
         this.limit = tableHandle.getLimit();
         this.fileModifiedTimeDomain = getFileModifiedTimePathDomain(tableHandle.getEnforcedPredicate());
+
+        // asynchronously collect the tasks and load them into the queue
+        this.queue = new ThrottledAsyncQueue<>(maxSplitsPerSecond, maxOutstandingSplits, executor);
```

Comment:
How many threads there can be?

Reply:
It uses the iceberg executor service for processing, which is created using `Executors.newCachedThreadPool`. That has 0 threads for corePoolSize and Integer.MAX_VALUE for maxPoolSize.